src/crypto/chacha_poly_aead.cpp

```diff
@@ -53,18 +85,24 @@ bool ChaCha20Poly1305AEAD::Crypt(uint64_t seqnr_payload, uint64_t seqnr_aad, int
 
     unsigned char expected_tag[POLY1305_TAGLEN], poly_key[POLY1305_KEYLEN];
     memset(poly_key, 0, sizeof(poly_key));
-    m_chacha_main.SetIV(seqnr_payload);
 
-    // block counter 0 for the poly1305 key
-    // use lower 32bytes for the poly1305 key
-    // (throws away 32 unused bytes (upper 32) from this ChaCha20 round)
-    m_chacha_main.Seek(0);
-    m_chacha_main.Crypt(poly_key, poly_key, sizeof(poly_key));
+    // 1. AAD (the encrypted packet length), use the header-keystream
+    if (is_encrypt) {
+        m_chacha_header.Crypt(src, dest, 3);
```

Comment:
The `ChaCha20Poly1305AEAD::Crypt` function uses argument order `dest`, `len`, `src`
The `ChaCha20Forward4064::Crypt` function uses argument order `input`, `output`, `bytes`
Let's standardize on one, please.

Reply:
Done. New implementation uses spans and I went with `input, output` everywhere.